icu4c/source/test/cintltst/ulocbuildertst.c

```diff
@@ -0,0 +1,1743 @@
+                  wellFormedAttributes[i], u_errorName(status));
+        }
+        if (i > 2) {
+            bld.removeUnicodeLocaleAttribute(wellFormedAttributes[i - 1]);
+            loc = bld.build(status);
+            if (U_FAILURE(status)) {
+                errln("removeUnicodeLocaleAttribute(\"%s\") got Error: %s\n",
+                      wellFormedAttributes[i - 1], u_errorName(status));
+            }
+            bld.removeUnicodeLocaleAttribute(wellFormedAttributes[i - 3]);
+            loc = bld.build(status);
+            if (U_FAILURE(status)) {
+                errln("removeUnicodeLocaleAttribute(\"%s\") got Error: %s\n",
+                      wellFormedAttributes[i - 3], u_errorName(status));
+            }
+        }
+    }
+}
+
+*/
+static void TestAddUnicodeLocaleAttributeIllFormed() {
+    static const char* illFormed[] = {
+        "aa",
+        "34",
+        "ab-",
+        "-cd",
+        "-ef-",
+        "zyzabcdef",
+        "123456789",
+        "ab-abc",
+        "1ZB30zfk9-abc",
+        "2ck30zfk9-adsf023-234kcZ",
+    };
+    for (int i = 0; i < UPRV_LENGTHOF(illFormed); i++) {
+        const char* ill = illFormed[i];
+        UErrorCode status = U_ZERO_ERROR;
+        ULocaleBuilder* bld = ulocbld_open();
+        ulocbld_addUnicodeLocaleAttribute(bld, ill, -1);
+        char buffer[ULOC_FULLNAME_CAPACITY];
+        ulocbld_buildLocaleID(bld, buffer, ULOC_FULLNAME_CAPACITY, &status);
+        if (status != U_ILLEGAL_ARGUMENT_ERROR) {
+            log_err("addUnicodeLocaleAttribute(\"%s\") should fail but has no Error\n",
+                  ill);
+        }
+        ulocbld_close(bld);
+    }
+}
+
+/*
+static void TestSetExtensionU() {
+    LocaleBuilder bld;
+    bld.setLanguage("zh");
+    Verify(bld, "zh",
+           "setLanguage(\"zh\") got Error: %s\n");
+
+    bld.setExtension('u', "co-stroke");
+    Verify(bld, "zh-u-co-stroke",
+           "setExtension('u', \"co-stroke\") got Error: %s\n");
+
+    bld.setExtension('U', "ca-islamic");
+    Verify(bld, "zh-u-ca-islamic",
+           "setExtension('U', \"zh-u-ca-islamic\") got Error: %s\n");
+
+    bld.setExtension('u', "ca-chinese");
+    Verify(bld, "zh-u-ca-chinese",
+           "setExtension('u', \"ca-chinese\") got Error: %s\n");
+
+    bld.setExtension('U', "co-pinyin");
+    Verify(bld, "zh-u-co-pinyin",
+           "setExtension('U', \"co-pinyin\") got Error: %s\n");
+
+    bld.setRegion("TW");
+    Verify(bld, "zh-TW-u-co-pinyin",
+           "setRegion(\"TW\") got Error: %s\n");
+
+    bld.setExtension('U', "");
+    Verify(bld, "zh-TW",
+           "setExtension('U', \"\") got Error: %s\n");
+
+    bld.setExtension('u', "abc-defg-kr-face");
+    Verify(bld, "zh-TW-u-abc-defg-kr-face",
+           "setExtension('u', \"abc-defg-kr-face\") got Error: %s\n");
+
+    bld.setExtension('U', "ca-japanese");
+    Verify(bld, "zh-TW-u-ca-japanese",
+           "setExtension('U', \"ca-japanese\") got Error: %s\n");
+
+}
+*/
```

Comment:
Why commented out?

Reply:
I am still working on them since the API is not yet approved.

Reply:
ready to review now.